crates/ruff_python_parser/src/python.lalrpop

```diff
@@ -1308,52 +1308,52 @@ Decorator: ast::Decorator = {
     },
 };
 
-YieldExpr: ast::Expr = {
+YieldExpr: ast::ParenthesizedExpr = {
     <location:@L> "yield" <value:TestList?> <end_location:@R> => ast::Expr::Yield(
-        ast::ExprYield { value: value.map(Box::new), range: (location..end_location).into() }
+        ast::ExprYield { value: value.map(ast::Expr::from).map(Box::new), range: (location..end_location).into() }
     ),
-    <location:@L> "yield" "from" <e:Test<"all">> <end_location:@R> => ast::Expr::YieldFrom(
-        ast::ExprYieldFrom { value: Box::new(e), range: (location..end_location).into() }
+    <location:@L> "yield" "from" <value:Test<"all">> <end_location:@R> => ast::Expr::YieldFrom(
+        ast::ExprYieldFrom { value: Box::new(value.into()), range: (location..end_location).into() }
     ),
 };
 
-Test<Goal>: ast::Expr = {
+Test<Goal>: ast::ParenthesizedExpr = {
     <location:@L> <body:OrTest<"all">> "if" <test:OrTest<"all">> "else" <orelse:Test<"all">> <end_location:@R> => ast::Expr::IfExp(
         ast::ExprIfExp {
-            test: Box::new(test),
-            body: Box::new(body),
-            orelse: Box::new(orelse),
+            test: Box::new(test.into()),
+            body: Box::new(body.into()),
+            orelse: Box::new(orelse.into()),
```

Comment:
We mostly need to add these `From`/`Into` conversions everywhere to translate between `ParenthesizedExpr` and `Expr`.